ext/session/session.c

```diff
@@ -1686,15 +1700,56 @@ static PHP_FUNCTION(session_set_cookie_params)
 		RETURN_FALSE;
 	}
 
-	convert_to_string_ex(lifetime);
+	if (Z_TYPE_P(lifetime_or_options) == IS_ARRAY) {
+		int found = 0;
+		zend_string *key;
+		zval *value;
+
+		ZEND_HASH_FOREACH_STR_KEY_VAL(Z_ARRVAL_P(lifetime_or_options), key, value) {
+			if (key) {
+				ZVAL_DEREF(value);
+				if(!strcasecmp("lifetime", ZSTR_VAL(key))) {
+					lifetime = value;
+					convert_to_string_ex(lifetime);
```

Comment:
This is going to modify the array in-place. Either this should have a strict type check, or use `zval_get_string` (with appropriate refcount management for the other case).